flink-table/flink-table-planner/src/main/java/org/apache/flink/table/sqlexec/SqlExecutableStatements.java

```diff
@@ -18,62 +18,197 @@
 	 * is subclass of {@code SqlNode}.
 	 *
 	 * <p>Caution that the {@link #execute(SqlNode)} should never expect to be invoked.
 	 *
-	 * @param tableEnvironment TableEnvironment to interact with
+	 * @param flinkPlanner     FlinkPlannerImpl to convert sql node to rel node
 	 * @param sqlNode          SqlNode to execute on
 	 */
-	public static void executeSqlNode(TableEnvironment tableEnvironment, SqlNode sqlNode) {
-		SqlExecutableStatements statement = new SqlExecutableStatements(tableEnvironment);
-		statement.dispatcher.invoke(sqlNode);
+	public static Operation executeSqlNode(FlinkPlannerImpl flinkPlanner, SqlNode sqlNode) {
+		// validate the query
+		final SqlNode validated = flinkPlanner.validate(sqlNode);
+		SqlExecutableStatements statement = new SqlExecutableStatements(flinkPlanner);
+		try {
+			return statement.dispatcher.invoke(validated);
+		} catch (RuntimeException r) {
+			// try to make the throws more friendly.
+			if (r.getCause() instanceof InvocationTargetException) {
+				final Throwable throwable = r.getCause().getCause();
+				if (throwable instanceof ValidationException) {
+					throw new ValidationException(throwable.getMessage());
+				} else if (throwable instanceof AssertionError) {
+					throw new AssertionError(throwable.getMessage());
+				} else {
+					throw new RuntimeException(throwable);
+				}
+			} else {
+				throw r;
+			}
+		}
 	}
 
 	/**
 	 * Execute the {@link SqlCreateTable} node.
 	 */
-	public void execute(SqlCreateTable sqlCreateTable) {
-		// need to implement.
+	public Operation execute(SqlCreateTable sqlCreateTable) {
+		// set with properties
+		SqlNodeList propertyList = sqlCreateTable.getPropertyList();
+		Map<String, String> properties = new HashMap<>();
+		if (propertyList != null) {
+			propertyList.getList().forEach(p ->
+				properties.put(((SqlProperty) p).getKeyString().toLowerCase(),
+					((SqlProperty) p).getValueString()));
+		}
+		String tableType = properties.get("connector");
+		if (StringUtils.isNullOrWhitespaceOnly(tableType)) {
+			throw new SqlExecutionException("Option [connector] is required for"
+				+ " CREATE TABLE statement.");
+		}
+
+		TableSchema tableSchema = createTableSchema(sqlCreateTable,
+			new FlinkTypeFactory(new FlinkTypeSystem())); // need to make type factory singleton ?
+		String tableComment = "";
+		if (sqlCreateTable.getComment() != null) {
+			tableComment = sqlCreateTable.getComment().getNlsString().getValue();
+		}
+		// set partition key
+		List<String> partitionKeys = new ArrayList<>();
+		SqlNodeList partitionKey = sqlCreateTable.getPartitionKeyList();
+		if (partitionKey != null) {
+			partitionKeys = partitionKey
+				.getList()
+				.stream()
+				.map(p -> ((SqlIdentifier) p).getSimple())
+				.collect(Collectors.toList());
+		}
+		CatalogTable catalogTable = new CatalogTableImpl(tableSchema,
+			partitionKeys,
+			properties,
+			tableComment);
+		return new CreateTableOperation(sqlCreateTable.fullTableName(), catalogTable, false);
```

Comment:
`SqlCreate` from calcite has such information for the boolean flag `ignoreIfExists`

Reply:
We actually hard code it to be false when create the `SqlCreateTable` instance, are we planning to support this now ?